InnerEye/ML/run_ml.py

```diff
@@ -208,75 +273,117 @@ def set_run_tags_from_parent(self) -> None:
+                if should_wait_for_other_child_runs:
+                    self.wait_for_runs_to_finish()
+                    self.create_ensemble_model_and_run_inference()
+        else:
+            # Inference for all models that are specified via LightningContainers
+            self.run_inference_for_lightning_models(checkpoint_handler.get_checkpoints_to_test(), trainer)
+
+    def run_inference_for_lightning_models(self, checkpoint_paths: List[Path], trainer: Optional[Trainer]) -> None:
+        """
+        Run inference on the test set for all models that are specified via a LightningContainer.
+        """
+        if len(checkpoint_paths) != 1:
+            raise ValueError(f"This method expects exactly 1 checkpoint for inference, but got {len(checkpoint_paths)}")
+        lightning_model = self.container.model
+        # Run the customized inference code only if the the "inference" step has been overridden
+        if isinstance(lightning_model, LightningInference) and \
+                type(lightning_model).inference_step != LightningInference.inference_step:
+            logging.info("Running inference via the LightningInference.inference_step method")
+            data = self.container.get_inference_data_module()
+            dataloaders: List[Tuple[DataLoader, ModelExecutionMode]] = []
+            if self.container.perform_validation_and_test_set_inference:
+                dataloaders.append((data.test_dataloader(), ModelExecutionMode.TEST))
+                dataloaders.append((data.val_dataloader(), ModelExecutionMode.VAL))
+            if self.container.perform_training_set_inference:
+                dataloaders.append((data.train_dataloader(), ModelExecutionMode.TRAIN))
+            map_location = "gpu" if self.container.use_gpu else "cpu"
+            checkpoint = pl_load(checkpoint_paths[0], map_location=map_location)
+            lightning_model.load_state_dict(checkpoint['state_dict'])
+            lightning_model.eval()
+            lightning_model.on_inference_start()
+            for loader, split in dataloaders:
+                logging.info(f"Starting inference on {split.value} set")
+                lightning_model.on_inference_epoch_start(dataset_split=split, is_ensemble_model=False)
+                for batch_idx, item in enumerate(loader):
+                    model_output = lightning_model.forward(item[0])
+                    lightning_model.inference_step(item, batch_idx, model_output=model_output)
+                lightning_model.on_inference_epoch_end()
+            lightning_model.on_inference_end()
+        elif type(lightning_model).test_step != LightningModule.test_step:
+            # Run Lightning's built-in test procedure if the `test_step` method has been overridden
+            logging.info("Running inference via the LightningModule.test_step method")
+            trainer = trainer or create_lightning_trainer(self.container)
```

Comment:
create_lightning_trainer returns a tuple, not just a Trainer

Reply:
GOOD CATCH! Thank you